src/index.ts

```diff
@@ -44,31 +52,85 @@ import {
 
 import '../style/index.css';
 
+
+/**
+ * A class that tracks editor widgets.
+ */
+export
+interface IPDFJSTracker extends IInstanceTracker<PDFJSViewer> {}
+
+
+/* tslint:disable */
+/**
+ * The editor tracker token.
+ */
+export
+const IPDFJSTracker = new Token<IPDFJSTracker>('@jupyterlab/latex:IPDFJSTracker');
+/* tslint:enable */
+
+const latexPluginId = '@jupyterlab/latex:plugin';
+
 namespace CommandIDs {
-  export const openLatexPreview = 'latex:open-preview';
+  /**
+   * Open a live preview for a `.tex` document.
+   */
+  export
+  const openLatexPreview = 'latex:open-preview';
+
+  /**
+   * Reveal in the editor a position from the pdf using SyncTeX.
+   */
+  export
+  const synctexEdit = 'latex:synctex-edit';
+
+  /**
+   * Reveal in the pdf a position from the editor using SyncTeX.
+   */
+  export
+  const synctexView = 'latex:synctex-view';
 }
+
+/**
+ * The options for a SyncTeX view command,
+ * mapping the editor position the PDF.
+ */
+type ISynctexViewOptions = CodeEditor.IPosition;
+
+/**
+ * The options for a SyncTeX edit command,
+ * mapping the pdf position to an editor position.
+ */
+type ISynctexEditOptions = PDFJSViewer.IPosition;
+
 /**
  * The JupyterLab plugin for the LaTeX extension.
  */
 const latexPlugin: JupyterLabPlugin<void> = {
-  id: 'jupyterlab-latex:open',
-  requires: [IDocumentManager, IEditorTracker, ILayoutRestorer, IStateDB],
+  id: latexPluginId,
+  requires: [
+    IDocumentManager,
+    IEditorTracker,
+    ILayoutRestorer,
+    IPDFJSTracker,
+    ISettingRegistry,
+    IStateDB
+  ],
   activate: activateLatexPlugin,
   autoStart: true
 };
 
 /**
  * Make a request to the notebook server LaTeX endpoint.
  *
- * @param url - the path to the .tex file to watch.
+ * @param path - the path to the .tex file to watch.
  *
  * @param settings - the settings for the current notebook server.
  *
- * @returns a Promise resolved with the JSON response.
+ * @returns a Promise resolved with the text response.
```

Comment:
Why did you switch this to text instead of JSON? was it just incorrectly documented? Because as we said, there may be good reason to send back more than just text if we ever need to send back metadata about the build.

Reply:
Yeah, I considered this an incorrect docstring: we are just sending back a text body, not a JSON-formatted response. If we want to send back more metadata, we should probably switch back to JSON.